osu.Desktop/DiscordRichPresence.cs

```diff
@@ -160,7 +219,27 @@ private string truncate(string str)
             });
         }
 
-        private int? getBeatmapID(UserActivity activity)
+        private static bool tryParseRoomSecret(ReadOnlySpan<char> secret, out long roomId, out string? password)
+        {
+            roomId = 0;
+            password = null;
+
+            int roomSecretSplitIndex = secret.IndexOf(':');
+
+            if (roomSecretSplitIndex == -1)
+                return false;
+
+            if (!long.TryParse(secret[..roomSecretSplitIndex], out roomId))
+                return false;
+
+            // just convert to string here, we're going to have to alloc it later anyways
```

Comment:
Not sure what's to be expected here.
```suggestion
```

Reply:
I don't remember what I was on about... :sweat_smile: